x-pack/plugins/security_solution/public/timelines/components/timeline/properties/use_notes_in_flyout.ts

```diff
@@ -0,0 +1,99 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { useCallback, useMemo, useState } from 'react';
+import { useDispatch } from 'react-redux';
+import { useDeepEqualSelector } from '../../../../common/hooks/use_selector';
+import { appSelectors } from '../../../../common/store';
+import { timelineActions } from '../../../store';
+
+interface UseNotesInFlyoutArgs {
+  eventIdToNoteIds: Record<string, string[]>;
+  refetch?: () => void;
+  timelineId: string;
+}
+
+const EMPTY_STRING_ARRAY: string[] = [];
+
+function isNoteNotNull<T>(note: T | null): note is T {
+  return note !== null;
+}
+
+export const useNotesInFlyout = (args: UseNotesInFlyoutArgs) => {
+  const [isNotesFlyoutVisible, setIsNotesFlyoutVisible] = useState(false);
+
+  const [eventId, setNotesEventId] = useState<string>();
+
+  const closeNotesFlyout = useCallback(() => {
+    setIsNotesFlyoutVisible(false);
+  }, []);
+
+  const showNotesFlyout = useCallback(() => {
+    setIsNotesFlyoutVisible(true);
+  }, []);
+
+  const { eventIdToNoteIds, refetch, timelineId } = args;
+
+  const getNotesByIds = useMemo(() => appSelectors.notesByIdsSelector(), []);
+
+  const notesById = useDeepEqualSelector(getNotesByIds);
+
+  const dispatch = useDispatch();
+
+  const noteIds: string[] = useMemo(
+    () => (eventIdToNoteIds && eventId && eventIdToNoteIds[eventId]) || EMPTY_STRING_ARRAY,
```

Comment:
nit: `eventIdToNoteIds && eventId ? eventIdToNoteIds[eventId]) : EMPTY_STRING_ARRAY`

Reply:
I remeber doing this but looks like it got reverted during many merges. Done here : 801c31448a2